packages/clerk-js/src/ui/elements/Select.tsx

```diff
@@ -0,0 +1,316 @@
+        <Flex
+          ref={floating}
+          onKeyDown={onKeyDown}
+          direction='col'
+          justify='start'
+          sx={[
+            theme => ({
+              backgroundColor: colors.makeSolid(theme.colors.$colorBackground),
+              border: theme.borders.$normal,
+              borderRadius: theme.radii.$lg,
+              borderColor: theme.colors.$blackAlpha200,
+              overflow: 'hidden',
+              width: '100%',
+              top: `calc(100% + ${theme.space.$2})`,
+              animation: `${animations.dropdownSlideInScaleAndFade} ${theme.transitionDuration.$slower} ${theme.transitionTiming.$slowBezier}`,
+              transformOrigin: 'top center',
+              boxShadow: theme.shadows.$cardDropShadow,
+              zIndex: 10,
+            }),
+            sx,
+          ]}
+          style={{ ...styles, left: styles.left - 1 }}
+          {...rest}
+        >
+          {searchOptions && (
+            <SelectSearchbar
+              placeholder={searchOptions.placeholder}
+              {...searchInputProps}
+            />
+          )}
+          <SelectOptionsContainer>
+            {filteredItems.map((option, index) => {
+              const isSelected = index === selectedIndex;
+              return (
+                <SelectOptionBuilder
+                  key={index}
+                  index={index}
+                  ref={isSelected ? selectedItemRef : undefined}
+                  option={option}
+                  optionBuilder={optionBuilder}
+                  isSelected={isSelected}
+                  handleSelect={select}
+                />
+              );
+            })}
+            {searchOptions && filteredItems.length === 0 && <SelectNoResults>{noResultsMessage}</SelectNoResults>}
+          </SelectOptionsContainer>
+        </Flex>
+      )}
+    </>
+  );
+};
+
+type SelectOptionBuilderProps<T> = {
+  option: T;
+  index: number;
+  optionBuilder: OptionBuilder<T>;
+  handleSelect: (option: T) => void;
+  isSelected: boolean;
+};
+
+const _SelectOptionBuilder = <T extends DefaultOption>(
+  props: SelectOptionBuilderProps<T>,
+  ref?: React.ForwardedRef<HTMLDivElement>,
+) => {
+  const { option, optionBuilder, index, handleSelect, isSelected } = props;
+  return (
+    <Flex
+      ref={ref}
+      sx={{
+        userSelect: 'none',
+      }}
+      onClick={() => {
+        handleSelect(option);
+      }}
+    >
+      {optionBuilder(option, index, isSelected)}
+    </Flex>
+  );
+};
+
+const SelectOptionBuilder = React.memo(React.forwardRef(_SelectOptionBuilder)) as <T extends DefaultOption>(
+  props: SelectOptionBuilderProps<T> & { ref?: React.ForwardedRef<HTMLDivElement> },
+) => ReturnType<typeof _SelectOptionBuilder>;
+
+const SelectSearchbar = (props: PropsOfComponent<typeof InputWithIcon>) => {
+  const { sx, ...rest } = props;
+  React.useEffect(() => {
+    // @ts-expect-error
+    return () => props.onChange({ target: { value: '' } });
```

Comment:
❓ Can you please elaborate on this line? Why do we trigger the `onChange` on unmount?

Reply:
We clear the search state provided by the `useSearchInput` hook to avoid persisting the search state when reopening. Do we not want this? It is the previous behaviour.